packages/circuit-ui/components/SideNavigation/components/PrimaryLink/PrimaryLink.tsx

```diff
@@ -0,0 +1,255 @@
+    margin-bottom: ${theme.spacings.kilo};
+  }
+`;
+
+const anchorActiveStyles = ({ theme, isActive }: StyleProps & AnchorProps) =>
+  isActive &&
+  css`
+    svg {
+      color: ${theme.colors.p500};
+    }
+
+    ${theme.mq.giga} {
+      color: ${theme.colors.p500};
+
+      &:hover {
+        background-color: ${theme.colors.p100};
+      }
+    }
+  `;
+
+const anchorOpenStyles = ({ theme, isOpen }: StyleProps & AnchorProps) =>
+  isOpen &&
+  css`
+    ${theme.mq.untilGiga} {
+      &::after {
+        right: 0;
+        left: 0;
+        width: 100%;
+      }
+    }
+  `;
+
+const anchorBadgeStyles = ({ theme, hasBadge }: StyleProps & AnchorProps) =>
+  hasBadge &&
+  css`
+    &::before {
+      display: block;
+      content: '';
+      position: absolute;
+      top: 14px;
+      left: 47px;
+      background-color: ${theme.colors.v500};
+      width: 10px;
+      height: 10px;
+      border-radius: ${theme.borderRadius.circle};
+    }
+
+    ${theme.mq.giga} {
+      &::before {
+        top: 6px;
+        left: 32px;
+      }
+    }
+  `;
+
+const Anchor = styled('a', {
+  shouldForwardProp: (prop) => isPropValid(prop),
+})<AnchorProps>(
+  focusVisible('inset'),
+  anchorStyles,
+  anchorActiveStyles,
+  anchorOpenStyles,
+  anchorBadgeStyles,
+);
+
+const iconStyles = (theme: Theme) => css`
+  flex-shrink: 0;
+  width: ${theme.iconSizes.mega};
+  height: ${theme.iconSizes.mega};
+  margin-right: ${theme.spacings.kilo};
+`;
+
+const suffixStyles = (theme: Theme) => css`
+  flex-shrink: 0;
+  width: ${theme.iconSizes.kilo};
+  height: ${theme.iconSizes.kilo};
+  margin-left: auto;
+  transition: transform ${theme.transitions.default};
+`;
+
+const externalIconStyles = css`
+  transform: rotate(-45deg);
+`;
+
+const labelStyles = ({ theme }: StyleProps) => css`
+  max-width: 100%;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
+
+  ${theme.mq.untilGiga} {
+    font-size: 24px;
+    line-height: 48px;
+  }
+
+  ${theme.mq.giga} {
+    max-width: 160px; /* max-width - icon-width - margin-right */
+  }
+`;
+
+const Label = styled(Body)(labelStyles);
+
+export function PrimaryLink({
+  icon: Icon,
+  label,
+  onClick,
+  isActive,
+  isOpen,
+  isExternal,
+  suffix: Suffix,
+  badge,
+  tracking,
+  secondaryGroups,
+  ...props
+}: PrimaryLinkProps): JSX.Element {
+  const { Link } = useComponents();
+
+  const handleClick = useClickEvent<ClickEvent>(
+    onClick,
+    tracking,
+    'primary-link',
+  );
+
+  const badgeId = badge && uniqueId();
+  const suffix = Suffix && <Suffix css={suffixStyles} role="presentation" />;
+  const isExternalLink = isExternal || props.target === '_blank';
+
+  return (
+    <Anchor
+      {...props}
+      onClick={handleClick}
+      isActive={isActive}
+      isOpen={isOpen}
+      hasBadge={Boolean(badge)}
+      aria-current={isActive ? 'page' : undefined}
+      // @ts-expect-error The type for the `as` prop is missing in Emotion's prop types.
```

Comment:
This is solved in 11 right?

Reply:
Yes, I made a PR for it which shipped with v11 😎

Reply:
🆒